File: src/static.cpp
#include <iostream>
using namespace std;

class Obj {
public:
    static int objNum;
    int attr;
    Obj() {
        ++objNum;
    }

    static void print() {
        cout << "objNum: " << Obj::objNum << endl; //Đúng
        // cout << "attr" << attr << endl; //Sai
    }
};

int Obj::objNum = 0;

int main() {
    Obj obj1;
    cout << Obj::objNum << endl;
    Obj obj2;
    cout << Obj::objNum << endl;
    Obj obj3;
    cout << Obj::objNum << endl;

    obj3.objNum = 10; // Cố tình sử dụng;
    cout << "Class Obj: " << Obj::objNum << endl;
    cout << "obj3: " << obj3.objNum << endl;
    return 0;
}

File: src/static.py
class Obj:
    objNum = 0 #static attribute
    def __init__(self):
        Obj.objNum += 1

class Math:
    __PI = 3.14
    
    @classmethod
    def getPI(cls):
        return cls.__PI

    @staticmethod
    def isBigger(a, b):
        return a > b

obj1 = Obj()
print(Obj.objNum)
obj2 = Obj()
print(Obj.objNum)
obj3 = Obj()
print(Obj.objNum)

obj3.objNum = 10
print("Class Obj:", Obj.objNum)
print("obj3:", obj3.objNum)

Obj.objNum = 7
print("Class Obj:", Obj.objNum)
print("obj3:", obj3.objNum)

print(Math.getPI())
print(Math.isBigger(4, 5))

